src/main/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/service/NextReviewDateService.kt

```diff
@@ -7,37 +7,50 @@ private const val BASIC = "Basic"
 
 data class NextReviewDateInput(
   val iepDetails: List<IepDetail>,
+  val hasAcctOpen: Boolean,
 )
 
-class NextReviewDateService {
+class NextReviewDateService(private val input: NextReviewDateInput) {
 
-  fun calculate(input: NextReviewDateInput): LocalDate {
-    val (iepDetails) = input
-    val lastReviewDate = iepDetails[0].iepDate
-
-    if (iepDetails[0].iepLevel == BASIC) {
-      return nextReviewDateForBasic(input)
+  fun calculate(): LocalDate {
+    if (isOnBasic()) {
+      return rulesForBasic()
     }
 
-    return lastReviewDate.plusYears(1)
+    return lastReviewDate().plusYears(1)
   }
 
-  private fun nextReviewDateForBasic(input: NextReviewDateInput): LocalDate {
-    val (iepDetails) = input
-    val lastReviewDate = iepDetails[0].iepDate
-
-    val lastReviewLevel = iepDetails[0].iepLevel
-    if (lastReviewLevel != BASIC) {
-      throw IllegalArgumentException("Programming error: private method nextReviewDateForBasic() called when lastReviewLevel was $lastReviewLevel")
-    }
-
+  private fun rulesForBasic(): LocalDate {
     // "if not suitable to return to Standard level further reviews must be undertaken at least every 28 days thereafter"
-    if (iepDetails.size >= 2 && iepDetails[1].iepLevel == BASIC) {
-      // IEP level was 'Basic' for last 2 reviews
-      return lastReviewDate.plusDays(28)
+    if (wasConfirmedBasic()) {
+      // "Exception for those identified as at Risk of Suicide and Self-harm (ACCT) and for young people,
+      // where further reviews must be undertaken at least every 14 days thereafter"
+      if (input.hasAcctOpen) {
+        return lastReviewDate().plusDays(14)
+      }
+
+      return lastReviewDate().plusDays(28)
     }
 
     // "PF 5.16 Prisoners placed on Basic must be reviewed within 7 days"
-    return lastReviewDate.plusDays(7)
+    return lastReviewDate().plusDays(7)
+  }
+
+  private fun lastReview(): IepDetail {
+    return input.iepDetails.first()
+  }
+
+  private fun isOnBasic(): Boolean {
+    return lastReview().iepLevel == BASIC
+  }
+
+  private fun lastReviewDate(): LocalDate {
+    return lastReview().iepDate
+  }
+
+  private fun wasConfirmedBasic(): Boolean {
```

Comment:
Is _Confirmed_ wording from the policy? If not could name this something like `lastTwoReviewsOnBasic`?

Reply:
Policy says "if _not suitable to return_ to Standard level", where "return to Standard" means get to an higher level.

The "not suitable to returns" means they were on 'Basic' but they were reviewed again and confirmed they would stay at the same level.

It's not wording from the policy, but regarding the method name I didn't want to use a name which was too low-level/tied to the last reviews being on basic even tho this is what about, hence the "wasConfirmedBasic" name. If it makes sense.

Reply:
e.g. I named it after the concept rather than the actual implementation/logic (or I would have called it `hasAtLeastTwoReviewsAndTheyAreBasicLevel()` or something 😝)

The idea is that if you really are curious about the implementation you can read the method body but when you're going through the calculation logic you only care about the high level concept.

Reply:
It's a private method anyway, if policy had a short word for the concept it would be nice to use but otherwise we could almost call this anything :)

Reply:
Makes sense 😄